explainer.md

```diff
@@ -63,7 +63,7 @@ The basic steps most WebXR applications will go through are:
 
 The first thing that any XR-enabled page will want to do is request an `XRDevice` and, if one is available, advertise VR functionality to the user. (For example, by adding a button to the page that the user can click to start XR content.)
 
-`navigator.xr.requestDevice` returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that resolves to an `XRDevice` if one is available. If no `XRDevice` is available, it will reject with a `NotFoundError`. The promise will also be rejected if an error occurs, such as the page not having the appropriate permissions to access VR capabilities.
+`navigator.xr.requestDevice` returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) that resolves to an `XRDevice` if one is available. If no `XRDevice` is available, it will reject with a "NotFoundError". The promise will also be rejected with an appropriate error if an error occurs during the device query. For example, if the page does not have the appropriate permissions to access XR capabilities it would reject with a "NotAllowedError".
```

Comment:
per my earlier comment, actually, it would be a `SecurityError` (see [Web Bluetooth's `requestDevice`](https://webbluetoothcg.github.io/web-bluetooth/#requestDevice-user-gesture) for measure)

Reply:
So I looked at that, and the case where they throw a `SecurityError` is when the call isn't triggered by user activation. (`requestDevice` in our API does not have that requirement.) Given the [description of the two errors](https://heycam.github.io/webidl/#securityerror):

**"SecurityError":** The operation is insecure.
**"NotAllowedError":** The request is not allowed by the user agent or the platform in the current context, possibly because the user denied permission.

It seems pretty explicit that a denied permission should use a `NotAllowedError`, while calling a function that requires user activation outside of user activation would produce a `SecurityError`.

Reply:
awesome, thanks for digging into this. I learned a few things :)